File: web/utils/num-utils.dart
// NetTango Copyright (C) Michael S. Horn, Uri Wilensky, and Corey Brady. https://github.com/NetTango/NetTango

part of NetTango;

class NumUtils {

  static int toInt(var d, [ int defalutValue = 0 ]) {
    if (d == null) {
      return defalutValue;
    }
    else if (d is int) {
      return d;
    }
    else if (d is String) {
      try {
        return int.parse(d);
      } on Exception {
        return defalutValue;
      }
    }
    return defalutValue;
  }

  static num toNum(var d, [ num defalutValue = 0 ]) {
    if (d == null) {
      return defalutValue;
    }
    else if (d is num) {
      return d;
    }
    else if (d is String) {
      try {
        return num.parse(d);
      } on Exception {
        return defalutValue;
      }
    }
    return defalutValue;
  }

  static int sum(Iterable<int> values) {
    return values.reduce( (a, b) => a + b );
  }

}

File: src/utils/num-utils.ts
// NetTango Copyright (C) Michael S. Horn, Uri Wilensky, and Corey Brady. https://github.com/NetTango/NetTango

class NumUtils {

  static toInt(d: any, defaultValue: number = 0): number {
    return Math.floor(NumUtils.toNum(d, defaultValue))
  }

  static toNum(d: any, defaultValue: number = 0): number {
    const maybeNum = NumUtils.toNumOrNull(d)
    return (maybeNum === null) ? defaultValue : maybeNum
  }

  static toNumOrNull(d: any) {
    if (d === null) {
      return null
    }
    switch (typeof(d)) {

      case "number":
        return d

      case "string":
        const result = Number.parseFloat(d)
        if (Number.isNaN(result)) {
          return null
        }
        return result

      default:
        return null

    }
  }

  static sum(values: number[]): number {
    return values.reduce( (a, b) => a + b )
  }

}
